engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/QueryStringPredicateOptionsStep.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.predicate.dsl;
+
+import java.util.function.Consumer;
+
+import org.hibernate.search.engine.search.common.Rewrite;
+
+/**
+ * The final step in a "query string" predicate definition, where optional parameters can be set.
+ *
+ * @param <S> The "self" type (the actual exposed type of this step).
+ */
+public interface QueryStringPredicateOptionsStep<S extends QueryStringPredicateOptionsStep<?>>
+		extends CommonQueryStringPredicateOptionsStep<S> {
+
+	/**
+	 * Specifies whether * and ? are allowed as first characters of a search term. Default is true.
+	 * @param allowLeadingWildcard
+	 * @return {@code this}, for method chaining.
+	 */
+	S allowLeadingWildcard(boolean allowLeadingWildcard);
+
+	//	/**
+	//	 * Specifies whether backend should attempt to analyze wildcard terms.
+	//	 * Default is false.
+	//	 *
+	//	 * @param analyzeWildcard
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S analyzeWildcard(boolean analyzeWildcard);
+
+	//	/**
+	//	 * Specifies whether to create a match phrase query automatically for multi-term synonyms. For example, if you specify ba, batting average as synonyms and search for ba, OpenSearch searches for ba OR "batting average" (if this option is true) or ba OR (batting AND average) (if this option is false). Default is true.
+	//	 * @param autoGenerateSynonymsPhraseQuery
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S autoGenerateSynonymsPhraseQuery(boolean autoGenerateSynonymsPhraseQuery);
+
+	/**
+	 * When true, resulting queries are aware of position increments.
+	 * This setting is useful when the removal of stop words leaves an unwanted “gap” between terms.
+	 * Default is true.
+	 *
+	 * @param enablePositionIncrements
+	 * @return {@code this}, for method chaining.
+	 */
+	S enablePositionIncrements(boolean enablePositionIncrements);
```

Comment:
Doesn't that require enabling term vectors or something?

Not sure it's worth it, but if you already implemented and tested it, why not... You'll need to document it though.